src/cpu/slime_check/simd.zig

```diff
@@ -0,0 +1,151 @@
+        self.seed = (self.seed *% multiplier +% addend) & mask;
+        return @intCast(self.seed >> @as(Vec64, @splat(48 - bits)));
+    }
+
+    /// Calculates `lanes` random numbers between 0 (inclusive) and `bound` (exclusive)
+    pub fn nextInts(self: *@This(), comptime bound: i32) Vec32 {
+        if (bound <= 0) @compileError("bound must be positive");
+
+        if (comptime std.math.isPowerOfTwo(bound)) {
+            return @intCast((bound * @as(i64, self.next(31))) >> 31);
+        }
+
+        const bounds: Vec32 = @splat(bound);
+
+        var bits: Vec32 = self.next(31);
+        var val: Vec32 = @mod(bits, bounds);
+        var biased = bits - val +% @as(Vec32, @splat(bound - 1)) < @as(Vec32, @splat(0));
+        while (@reduce(.Or, biased)) {
+            @setCold(true);
+            // @branchHint(.cold);
+
+            bits = self.next(31);
+            val = @select(i32, biased, @mod(bits, bounds), val);
+            biased = @bitCast(
+                @intFromBool(biased) &
+                    @intFromBool(bits - val +% @as(Vec32, @splat(bound - 1)) <
+                    @as(Vec32, @splat(0))),
+            );
+        }
+        return val;
+    }
+
+    /// Calculates `lanes` random numbers between 0 (inclusive) and `bound` (exclusive)
+    /// Skips bias correction, but very occasionally (<1 in 100,000,000) gives an incorrect result
+    pub fn nextIntsBiased(self: *@This(), comptime bound: i32) Vec32 {
+        if (bound <= 0) @compileError("bound must be positive");
+
+        if (comptime std.math.isPowerOfTwo(bound)) {
+            return @intCast((bound * @as(i64, self.next(31))) >> 31);
+        }
+
+        const bounds: Vec32 = @splat(bound);
+
+        const bits: Vec32 = self.next(31);
+        const val: Vec32 = @mod(bits, bounds);
+        return val;
+    }
+
+    test nextIntsBiased {
+        // We expect `nextIntsBiased` to be incorrect on this seed
+        const seed: i64 = 304837631;
+        var random1 = @This().init(@splat(seed));
+        var random2 = @This().init(@splat(seed));
+        try std.testing.expect(random1.nextInts(10)[0] != random2.nextIntsBiased(10)[0]);
+    }
+};
+
+comptime {
+    _ = Random;
+}
+
+test areSlime {
+    const test_block = @import("../test_data.zig").block;
+    for (test_block, 0..) |row, z| {
+        for (row, 0..) |c, x| {
+            try std.testing.expectEqual(c == 'O', areSlime(0x51133, @intCast(x), @intCast(z))[0]);
+        }
+    }
+
+    const random = @import("../test_data.zig").random;
+    for (random) |location| {
+        try std.testing.expectEqual(location.slime, areSlime(0x51133, location.x, location.z)[0]);
+    }
+}
+
+test "scalar simd parity" {
+    var rng = std.Random.DefaultPrng.init(0x51133);
+    const rand = rng.random();
+
+    for (0..10000) |_| {
+        const x = rand.intRangeAtMost(i32, -30_000_000, 30_000_000);
+        const z = rand.intRangeAtMost(i32, -30_000_000, 30_000_000);

```

Comment:
min/max chunk coords are 1.875M, not 30M :)

Reply:
oops :3